stellargraph/layer/graphsage.py

```diff
@@ -119,9 +119,300 @@ def call(self, x, **kwargs):
+        **kwargs
+    ):
+        self.output_dim = output_dim
+        assert output_dim % 2 == 0
+        self.half_output_dim = int(output_dim / 2)
+
+        # TODO: These should be user parameters
+        self.hidden_dim = output_dim
+        self.hidden_act = activations.get("relu")
+
+        self.has_bias = bias
+        self.act = activations.get(act)
+        self.w_neigh = None
+        self.w_self = None
+        self.w_pool = None
+        self.b_pool = None
+        self.bias = None
+        self._initializer = "glorot_uniform"
+        super().__init__(**kwargs)
+
+    def get_config(self):
+        """
+        Gets class configuration for Keras serialization
+
+        """
+        config = {
+            "output_dim": self.output_dim,
+            "bias": self.has_bias,
+            "act": activations.serialize(self.act),
+        }
+        base_config = super().get_config()
+        return {**base_config, **config}
+
+    def build(self, input_shape):
+        """
+        Builds layer
+
+        Args:
+            input_shape (list of list of int): Shape of input tensors for self
+            and neighbour
+
+        """
+        self.w_pool = self.add_weight(
+            name="w_pool",
+            shape=(input_shape[1][3], self.hidden_dim),
+            initializer=self._initializer,
+            trainable=True,
+        )
+        self.b_pool = self.add_weight(
+            name="b_pool",
+            shape=(self.hidden_dim,),
+            initializer=self._initializer,
+            trainable=True,
+        )
+        self.w_self = self.add_weight(
+            name="w_self",
+            shape=(input_shape[0][2], self.hidden_dim),
+            initializer=self._initializer,
+            trainable=True,
+        )
+        self.b_self = self.add_weight(
+            name="b_self",
+            shape=(self.hidden_dim,),
+            initializer=self._initializer,
+            trainable=True,
+        )
+        self.w_final = self.add_weight(
+            name="w_final",
+            shape=(2 * self.hidden_dim, self.output_dim),
+            initializer=self._initializer,
+            trainable=True,
+        )
+        if self.has_bias:
+            self.bias = self.add_weight(
+                name="bias",
+                shape=(self.output_dim,),
+                initializer="zeros",
+                trainable=True,
+            )
+        super().build(input_shape)
+
+    def call(self, x, **kwargs):
+        """
+        Apply max aggregation on input tensors, x
+
+        Args:
+          x: Keras Tensor
+
+        Returns:
+            Keras Tensor representing the aggregated embeddings in the input.
+
+        """
+        # x[0]: self vector (batch_size, head size, feature_size)
+        # x[1]: neighbour vector (batch_size, head size, neighbours, feature_size)
+        xw_neigh = K.dot(x[1], self.w_pool) + self.b_pool
+        xw_self = K.dot(x[0], self.w_self) + self.b_self
```

Comment:
I'd expect `MaxPoolAggregator` and `MeanPoolAggregator` to have similar structure, differing only in the `K.max` or `K.mean` operation applied to the neighbours' features. Here, the two classes are also different in parameters: for example, the `MeanPoolAggregator` class has `self.b_self bias`, which has no counterpart in the `MaxPoolAggregator` class.

Reply:
I've modified the meanpool class to act the same as the maxpool class.

Reply:
I've also renamed these aggregators to MeanPoolingAggregator and MaxPoolingAggregator